http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -663,23 +709,28 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
 
       auth <- samDAO.getLeoAuthToken
 
-      diskResourceOpt <- controlledResourceQuery
-        .getWsmRecordForRuntime(e.runtimeId, WsmResourceType.AzureDisk)
-        .transaction
-      _ <- diskResourceOpt.traverse { disk =>
-        // TODO: once we start supporting persistent disk, we should not delete disk anymore
-        wsmDao.deleteDisk(
-          DeleteWsmResourceRequest(
-            e.workspaceId,
-            disk.resourceId,
-            DeleteControlledAzureResourceRequest(
-              WsmJobControl(WsmJobId(s"delete-disk-${ctx.traceId.asString.take(10)}"))
+      deleteDiskAction =
```

Comment:
Hmm... semantically, I find it reads a bit weird to assign this to a variable and check `e.useExistingDisk` after. I would prefer to lead with `e.useExistingDisk match` here, so its clear `!e.useExistingDisk` means `delete new disk`. Maybe a comment saying we should only be deleting a disk that was just created as part of this VM creation too

Reply:
totally makes sense, I switched it to be a case and added a comment!